From f69621925beb36723225d2e77fd33af3751b89b6 Mon Sep 17 00:00:00 2001
From: Raphael Mounier <mounierr07@gmail.com>
Date: Sun, 21 May 2023 12:14:42 +0200
Subject: [PATCH 1/2] treble : add aosp charger 1/3

This patch allows to use the aosp charger. It replaces this commit "Add offline Charge Service" - TrebleDroid/platform_system_core@43f44ff

The source code of the charger service is declared in system/core/health
It manipulates leds, brightness and power, so it is necessary to add some selinux rules
---
 base.mk             | 11 +++++------
 charger.rc          | 17 +++++++++++++++++
 sepolicy/charger.te | 15 +++++++++++++++
 3 files changed, 37 insertions(+), 6 deletions(-)
 create mode 100644 charger.rc
 create mode 100644 sepolicy/charger.te

diff --git a/base.mk b/base.mk
index 348799e..2bf8089 100644
--- a/base.mk
+++ b/base.mk
@@ -49,7 +49,11 @@ PRODUCT_SYSTEM_DEFAULT_PROPERTIES += \
 #VNDK config files
 PRODUCT_COPY_FILES += \
 	device/phh/treble/vndk-detect:system/bin/vndk-detect \
-	device/phh/treble/vndk.rc:system/etc/init/vndk.rc \
+	device/phh/treble/vndk.rc:system/etc/init/vndk.rc
+
+#Charger config files
+PRODUCT_COPY_FILES += \
+	device/phh/treble/charger.rc:system/etc/init/charger.rc
 
 #USB Audio
 PRODUCT_COPY_FILES += \
@@ -84,11 +88,6 @@ PRODUCT_PACKAGES += \
 	bootctl \
 	vintf \
 
-# Fix Offline Charging on Huawmeme
-PRODUCT_PACKAGES += \
-	huawei-charger
-PRODUCT_COPY_FILES += \
-	$(call find-copy-subdir-files,*,device/phh/treble/huawei_charger/files,system/etc/charger)
 
 PRODUCT_COPY_FILES += \
 	device/phh/treble/twrp/twrp.rc:system/etc/init/twrp.rc \
diff --git a/charger.rc b/charger.rc
new file mode 100644
index 0000000..b52e93b
--- /dev/null
+++ b/charger.rc
@@ -0,0 +1,17 @@
+# Copyright (C) 2023 The TrebleDroid Project
+#
+#
+
+# Add vendor charger init
+import /vendor/etc/init/${ro.bootmode}/init.${ro.bootmode}.rc
+
+service charger /system/bin/charger
+    class charger
+    user root
+    group system root shell graphics input wakelock
+    capabilities SYS_BOOT
+    seclabel u:r:charger:s0
+
+on charger && property:ro.hardware=mt*
+    write /sys/class/leds/lcd-backlight/trigger "backlight"
+    write /sys/class/android_usb/android0/enable 1
diff --git a/sepolicy/charger.te b/sepolicy/charger.te
new file mode 100644
index 0000000..cbcb255
--- /dev/null
+++ b/sepolicy/charger.te
@@ -0,0 +1,15 @@
+# Allow charger to write to sysfs_backlight_attr (only for huawei)
+attribute sysfs_backlight_attr;
+allow charger sysfs_backlight_attr:file rw_file_perms;
+
+# Allow charger to write to sysfs_led_attr (only for huawei)
+# attribute sysfs_led_attr;
+# allow charger sysfs_led_attr:file rw_file_perms;
+
+# Allow charger to read and write to sysfs_power
+allow charger sysfs_power:file rw_file_perms;
+allow charger sysfs_power:dir r_dir_perms;
+allow charger sysfs_power:lnk_file read;
+
+# The system charger can write  powerctl properties
+set_prop(charger, powerctl_prop)
-- 
2.25.1

